Automattic/wp-calypso
# Stats: update checkout URL to site-only checkout

Related to pbNhbs-aEE-p2

## Proposed Changes

* This PR updates how the checkout URL is created to point it to the site-only checkout to avoid displaying the login screen before the user arrives at the checkout.

## Why are these changes being made?


* We are improving the checkout flow by removing a barrier when the user clicks to purchase a product from My Jetpack.

## Testing Instructions



* Point `widgets.wp.com` to your sandbox
* In your sandbox, run `bin/install-plugin.sh odyssey-stats add/checkout-url-handling-stats-in-my-jetpack`
* Spin up a JN site with access to your sandbox, or start your local Jetpack environment
* Make sure your Jetpack is just site-connected
* Go to `/wp-admin/admin.php?page=my-jetpack`
* Click on "Upgrade" in the Stats section
* <img width="1107" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5714212/489e1eef-74e2-4f5d-87ea-bac0600c61dd">
* After being redirected to the Stats interstitial, you can either click on "Contribute and continue" to go to the checkout, or click on "require a commercial license" - which will redirect you to the commercial interstitial - then click on "Upgrade and continue"
* <img width="300" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5714212/6fa17ab7-5ac5-4188-abb4-f641614c965c"> <img width="300" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5714212/6b305877-036e-4af7-afbb-bb29c04bd4d3">
* Confirm that you are redirected to the checkout page (without being prompted to log in first)

## Pre-merge Checklist
